synapse/lib/ast.py

```diff
@@ -3491,10 +3491,9 @@ async def run(self, runt, genr):
             raise s_stormctrl.StormReturn(valu)
 
         # no items in pipeline... execute
-        if self.kids:
+        if self.kids and self.isRuntSafe(runt):
```

Comment:
What is the return expected behavior if self.kids is true and self.isRuntSafe returns false?

Reply:
no-op